### ISSUE TITLE
UI translation

### ISSUE THREAD

**Issue by trollixx** [zealdocs/zeal#275]
Zeal needs to be translated in some popular languages.

Here's a list of top 10 languages (according to the today's statistics of zealdocs.org visitors, excluding English):
1. French
2. Simplified Chinese
3. Russian
4. Spanish
5. German
6. Portuguese (Brazil)
7. Japanese

Potentially (require better usage analytics):
- Polish
- Dutch
- Italian

An online translation system should be set up. Current options:
- Transifex
- OneSky
- Weblate

Perhaps, the list of supported languages should be kept small to ease maintaining of updated translations.

**Comment by Alkaid-Benetnash** [comment]
I want to help translate zeal's UI into Simplified Chinese. How can I help?

**Comment by trollixx** [comment]
Hi, nothing has been done for translation yet, so just keep an eye on this issue. I'll post an update once there's a translation system in place.

**Comment by JIghtuse** [comment]
Russian here. Ready to help when you choose a system.

**Comment by Horoneru** [comment]
French here. I will help whenever you got a translation system ready !

**Comment by asdrubalivan** [comment]
I will gladly help with Spanish, ping me when the translation system is in place!

**Comment by trollixx** [comment]
Thank you guys, I'll try to spend some time on the translation infrastructure soon.

**Comment by Horoneru** [comment]
You could try to use Qt's translation system? Then you send the file to be translated to the translator, he gives it to you, and you only need to update the translations.

**Comment by trollixx** [comment]
@Horoneru that way would cause additional headache when strings change (happens quite often). A translation platform would allow us to manage all changes easily and update translations accordingly. I guess I'll setup a SkyOne account or a Weblate instance. I am just trying to minimise my future work and make stuff automatic as much as possible.

Another reason why I am not very quick on translations is the fact, that all documentation is anyway in English. So, the translated UI wouldn't help people much. In my opinion software developers have to know English nowadays. :)

**Comment by TobiX** [comment]
Maybe hosted weblate? http://weblate.org/en/hosting/

**Comment by Kapeli** [comment]
> Another reason why I am not very quick on translations is the fact, that all documentation is anyway in English. So, the translated UI wouldn't help people much

Fwiw, this is also why Dash is only available in English. However, I do see how some users might prefer having all apps on their system in their native language, even if they use English docs. I haven't seen a huge demand for it though, got very few requests for it.

> In my opinion software developers have to know English nowadays

Some don't, but not many. Unfortunately it's very hard to support anything but English when it comes to docsets and the effort isn't worth it in my opinion.

**Comment by trollixx** [comment]
@TobiX yes, that's the main option I am considering right now. I need to figure out how well Weblate integrates with GitHub, but preliminary I really like it.

@Kapeli, thanks for your comments!

To all: I would really like to hear some strong arguments in favour of the translations support.

To my mind it might be good for Zeal to support several languages which have really large IT communities behind or cover many countries at once. My personal unordered list:
* Simplified Chinese (China is our main user base, not sure whether it also covers Taiwan);
* Russian (in addition to Russia covers ex-USSR countries like Belarus and Ukraine);
* Spanish (Spain + Latin America);
* Portuguese (Brazil + Portugal);
* French (France + Quebec);
* German (Germany itself has pretty big IT market).

The implementation itself is pretty simple, but the future maintenance is what I am concerned about.

**Comment by david-geiger** [comment]
Hi trollixx,

What is the status now of this enhancement report for internationalisation?

I'm volunteer for the French translation

**Comment by trollixx** [comment]
No progress at all yet :( Generating translation files should not be a problem, but keeping them up to date might be.

**Comment by david-geiger** [comment]
keeping them up to date is not a big problem, what you can do is to prevent by email, 4 or 5 days before out a new release, all translators that translation files have been updated and therefore can be translated in order to come out a new release in a few days.
This is what other developers do in their project.

**Comment by timarcosdias** [comment]
Hi everybody.

Let me explain you guys our situation here: I'm teaching programming for people aged from 13 years old in a small town where almost nobody really have basic English language skills. I know some will say developers have to know English, but it's unfortunately not even close to our reality here. My students are using Zeal to download the docs of the technologies we use in the classroom because they don't always have access to the Internet not even in our campus.

That said, I'm willing to help translate Zeal and make it possible for the users to select docsets in their native language when it's available. There's HTML, CSS, JavaScript and PHP docs in portuguese, for example, and it would be really helpful if the users could choose in which language they will download the docset.

Hope we can make it happen.

**Comment by trollixx** [comment]
@timarcosdias thanks for describing your situation. As I said before, translating Zeal itself is not a big problem, but that would unlikely do any good. Docsets themselves would need to be generated in different languages, and that is currently out of my control.

**Comment by timarcosdias** [comment]
@trollixx docsets provided by MDN (HTML, CSS and JavaScript), PHP docs and Python docs already have multilanguage support. Do you think it's possible for us to start from these?

**Comment by Kapeli** [comment]
Docsets are generated and indexed by me. There are currently only English versions of docsets. Please note that you can generate your own docsets - https://kapeli.com/docsets.

**Comment by trollixx** [comment]
@timarcosdias it is certainly possible to generate any kinds of docsets. Maintenance is the main question. All the docsets, you can currently download directly from Zeal, are coming from @Kapeli.

I guess the best option might be setting up generation for the Portugese versions of the few docsets you are interested in. See the link above for the instructions. Then they could be installed with the help of XML feeds.

Anyway, generating docsets in other languages is a whole new project, that is not directly connected to Zeal. And I barely have enough time to keep Zeal floating. But if you want to work on that, I'd be glad to provide guidance and assistance should you require any.

**Comment by timarcosdias** [comment]
Thanks @Kapeli and @trollixx .

I'll take some time to learn about the docsets generation and look for some friends to help in this project. I'll hopefully get in touch soon.

**Comment by sikmir** [comment]
> An online translation system should be set up. Current options:
>
>   Transifex
>    OneSky
>    Weblate

I would say Weblate is the best option here.

And ready to help with Russian translation.

**Comment by trollixx** [comment]
> I would say Weblate is the best option here.

That's what I was leaning towards in the past. In the meanwhile adding UI translations has very low priority.

**Comment by Diam395** [comment]
Hi, I think I've found a solution for translating documentation.
I downloaded a French version of the python documentation
in html format and copied the contents into C:\Users\hp\AppData\Local\Zeal\Zeal\docsets\Python_3.docset\Contents\Resources\Documents\doc
It works on Windows
